### PR TITLE
Add kainjow-mustache/4.1

### PR DESCRIPTION
**Pull request by syoliver**
Specify library name and version:  **kainjow-mustache/4.1**

Mustache implementation for modern C++ (requires C++11)

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/kainjow-mustache/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("mustache.hpp", dst=os.path.join("include", "kainjow", "mustache"), src=self._source_subfolder)
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "KainjowMustache"
```
Comment (uilianries) [inline recipes/kainjow-mustache/all/conanfile.py:34]: According the PR https://github.com/kainjow/Mustache/pull/37/files, the original name should be kainjow_mustache

WDYT @gocarlos ?

File: recipes/kainjow-mustache/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.9)
+project(test_package)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
```
Comment (uilianries) [inline recipes/kainjow-mustache/all/test_package/CMakeLists.txt:4]:
```suggestion
```

File: recipes/kainjow-mustache/all/conanfile.py (hunk 1 above)
Comment (syoliver) [reply recipes/kainjow-mustache/all/conanfile.py]: Renamed to kainjow_mustache

File: recipes/kainjow-mustache/all/test_package/CMakeLists.txt (hunk 2 above)
Comment (syoliver) [reply recipes/kainjow-mustache/all/test_package/CMakeLists.txt]: Done

**Comment by uilianries** [comment]
@SSE4 @danimtb @jgsogo Same old discussion about package name:

- kainjow_mustache
- kainjow-mustache

The CMake refers as kainjow_mustache. I kainjow-mustache as the convention for CCI, but it doesn't follow the upstream.

**Comment by jgsogo** [comment]
Maybe we can ask the author about it (here or in one issue in the origin repo). Nevertheless, I think `mustache` is not a common library name and maybe we don't need to use the prefix at all. Also, the only reference in sources I've found so far says it is `mustache` (https://github.com/kainjow/Mustache/blob/master/tests/CMakeLists.txt).

What about suing just `mustache`? wdyt?

**Comment by syoliver** [comment]
> What about suing just `mustache`? wdyt?

Mustache is the templating language, here it is the implementation of @kainjow. I don't know if there is another implementation with the same name in C++ but it may be possible. Then for barware compatibility I think it is better to name it kainjow_mustache, like nlohmann_json.
It namespace in this library is kainjow::mustache.

**Comment by jgsogo** [comment]
Oh! Didn't realize it was the name of the template system. Yes, then we should go for `kainjow-mustache` as we are doing for other libraries. And what about the CMake's find package module? `mustache::mustache`? `kainjow_mustache::kainjow_mustache`?

**Comment by gocarlos** [comment]
> kainjow_mustache::kainjow_mustache

same as nlohman_json

**Comment by Croydon** [comment]
I'm on the same page as @jgsogo here. It should be `kainjow-mustache` according to what is written down as our naming convention.

We also use this name in Bincrafters (and inexorgame before that) https://github.com/bincrafters/conan-kainjow-mustache

We already have some names diverging from this naming convention. Since it is not written down to have a look at some build system target names, this would be another exception to the rule

**Comment by uilianries** [comment]
> > kainjow_mustache::kainjow_mustache
> 
> same as nlohman_json

nlohman_json is explicit declared in the upstream, but here we don't have such declaration. Only mustache: https://github.com/kainjow/Mustache/blob/v4.1/CMakeLists.txt

**Comment by uilianries** [comment]
I agree with @Croydon , `kainjow-mustache` is better due CCI convention.

**Comment by syoliver** [comment]
> I agree with @Croydon , `kainjow-mustache` is better due CCI convention.

kainjow-mustache for the package name and kainjow_mustache::kainjow_mustache for the cmake target? Or kainjow::mustache with components?

**Comment by jgsogo** [comment]
> > I agree with @Croydon , `kainjow-mustache` is better due CCI convention.
> 
> kainjow-mustache for the package name and kainjow_mustache::kainjow_mustache for the cmake target? Or kainjow::mustache with components?

Even with components, Conan will generate a component with all the libraries inside the package, so, given your last alternative, Conan would generate `kainjow::kainjow` which is probably not what we want (we need something related to `mustache` there).

IMO, the alternatives are `kainjow_mustache::kainjow_mustache` or `mustache::mustache`.

File: recipes/kainjow_mustache/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+from conans import ConanFile, tools
+import os
+import glob
+
+
+class KainjowMustacheConan(ConanFile):
+    name = "kainjow_mustache"
```
Comment (Croydon) [inline recipes/kainjow_mustache/all/conanfile.py:7]:
```suggestion
    name = "kainjow-mustache"
```

**Pull request merged**